CONTRIBUTING.rst

```diff
@@ -0,0 +1,258 @@
+    $ git push
+
+#. Create a pull request from your branch to `main`.
+#. Once the pull request is merged, create a new release on GitHub. On the main branch, run::
+
+    $ git tag v0.2.0
+    $ git push --tags
+
+   This will trigger a GitHub workflow to build the package and upload it to TestPyPI. At the same time, the GitHub workflow will create a draft release on GitHub. Assuming that the workflow passes, the final release can then be published on GitHub by finalizing the draft release.
+
+#. Once the release is published, the `publish-pypi.yml` workflow will go into an `awaiting approval` mode on Github Actions. Only authorized users may approve this workflow (notifications will be sent) to trigger the upload to PyPI.
+
+.. warning::
+
+    Uploads to PyPI can **never** be overwritten. If you make a mistake, you will need to bump the version and re-release the package. If the package uploaded to PyPI is broken, you should modify the GitHub release to mark the package as broken, as well as yank the package (mark the version  "broken") on PyPI.
+
+Packaging
+---------
+
+When a new version has been minted (features have been successfully integrated test coverage and stability is adequate), maintainers should update the pip-installable package (wheel and source release) on PyPI as well as the binary on conda-forge.
+
+The simple approach
+~~~~~~~~~~~~~~~~~~~
+
+The simplest approach to packaging for general support (pip wheels) requires that ``flit`` be installed::
+
+    $ python -m pip install flit
+
+From the command line on your Linux distribution, simply run the following from the clone's main dev branch::
+
+    # To build the packages (sources and wheel)
+    $ python -m flit build
+
+    # To upload to PyPI
+    $ python -m flit publish dist/*
+
+The new version based off of the version checked out will now be available via `pip` (`$ pip install xncml`).
+
+Releasing on conda-forge
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Initial Release
+^^^^^^^^^^^^^^^
+
+Before preparing an initial release on conda-forge, we *strongly* suggest consulting the following links:
+ * https://conda-forge.org/docs/maintainer/adding_pkgs.html
+ * https://github.com/conda-forge/staged-recipes
+
+In order to create a new conda build recipe, to be used when proposing packages to the conda-forge repository, we strongly suggest using the ``grayskull`` tool::
+
+    $ python -m pip install grayskull
+    $ grayskull pypi xncml
+
+For more information on ``grayskull``, please see the following link: https://github.com/conda/grayskull
+
+Before updating the main conda-forge recipe, we echo the conda-forge documentation and *strongly* suggest performing the following checks:
+ * Ensure that dependencies and dependency versions correspond with those of the tagged version, with open or pinned versions for the `host` requirements.
+ * If possible, configure tests within the conda-forge build CI (e.g. `imports: xncml`, `commands: pytest xncml`).
```

Comment:
Consider some `make release-conda` or similar.

Reply:
I don't think it's necessary to create a `make release-conda` as the initial step for the conda-forge release requires manual step to create the conda-forge feedstock and then, when creating new releases, conda-forge should automatically get the pipy release and create a PR on the feedstock.

Reply:
In all honesty, I wrote this to provide some guidance for new maintainers. If it's irrelevant (i.e. the project was already on conda), there's no harm in removing it.

Perhaps this kind of information would be helpful in a separate documentation file **specific for maintainers creating the project from the cookiecutter template** that can be more easily removed?

Reply:
I'm fine either way if it feels overkill for this use case.

I personally prefer to push for the `make ...` approach because it allows abstraction (CI doesn't need updates if the tool changes) and to keep all the logic together in `Makefile` rather than spreading it around with custom tweak and combinations of various tool commands.

Reply:
@Zeitsperre I agree, I think this would fit better in a doc/development/release.rst document as it's rather the job of maintainers and not of the developers.

@fmigneault I also agree. However, one place where most tools are duplicated is the pre-commit config. 
I wonder if we can tell pre-commit to run a `make lint` and `make format` instead of it fetching these steps from other git repositories.
BTW, that's pretty much the approach of javascripts projects using husky, they define the lint, format, etc, commands once in a versioned package.json and in the pre-commit config of husky, they just call these commands.